pkg/gatewayserver/io/basicstationlns/basicstationlns.go

```diff
@@ -34,36 +36,64 @@ import (
 	"go.thethings.network/lorawan-stack/pkg/ttnpb"
 	"go.thethings.network/lorawan-stack/pkg/unique"
 	"go.thethings.network/lorawan-stack/pkg/web"
+	"go.thethings.network/lorawan-stack/pkg/web/middleware"
 	"google.golang.org/grpc/metadata"
 )
 
 var (
 	errEmptyGatewayEUI = errors.Define("empty_gateway_eui", "empty gateway EUI")
+	errListener        = errors.DefineFailedPrecondition(
+		"listener",
+		"failed to serve Basic Station frontend listener",
+	)
 )
 
 type srv struct {
-	ctx      context.Context
-	server   io.Server
-	upgrader *websocket.Upgrader
-	tokens   io.DownlinkTokens
+	ctx       context.Context
+	server    io.Server
+	webServer *echo.Echo
+	upgrader  *websocket.Upgrader
+	tokens    io.DownlinkTokens
 }
 
 func (*srv) Protocol() string { return "basicstation" }
 
-// New returns a new Basic Station frontend that can be registered in the web server.
-func New(ctx context.Context, server io.Server) web.Registerer {
+// Start starts the Basic Station front end.
+func Start(ctx context.Context, server io.Server, listener net.Listener) error {
+	webServer := echo.New()
+	webServer.Logger = web.NewNoopLogger()
+	webServer.Listener = listener
+	webServer.Use(
+		middleware.ID(""),
+		echomiddleware.BodyLimit("16M"),
+		middleware.Recover(),
+	)
+
 	ctx = log.NewContextWithField(ctx, "namespace", "gatewayserver/io/basicstation")
-	return &srv{
-		ctx:      ctx,
-		server:   server,
-		upgrader: &websocket.Upgrader{},
+	s := &srv{
+		ctx:       ctx,
+		server:    server,
+		upgrader:  &websocket.Upgrader{},
+		webServer: webServer,
 	}
+
+	webServer.GET("/router-info", s.handleDiscover)
+	webServer.GET("/:id", s.handleTraffic)
```

Comment:
Please use `/traffic/:id` still, otherwise we have to blacklist `router-info` as ID and we would be very limited in other stuff in future BS versions